src/PBR/Application/PBR/Shaders/spmap_cs.glsl

```diff
@@ -0,0 +1,174 @@
+#version 450 core
+// Physically Based Rendering
+// Copyright (c) 2017-2018 Michał Siejak
+
+// Pre-filters environment cube map using GGX NDF importance sampling.
+// Part of specular IBL split-sum approximation.
+
+const float PI = 3.141592;
+const float TwoPI = 2 * PI;
+const float Epsilon = 0.00001;
+
+const uint NumSamples = 1024;
+const float InvNumSamples = 1.0 / float(NumSamples);
+
+const int NumMipLevels = 1;
+layout(set=0, binding=0, rgba16f) restrict writeonly uniform image2DArray outputTexture[NumMipLevels];
+layout(set=0, binding=1) uniform textureCube inputTexture;
+layout(set=0, binding=2) uniform FakePushConstants {
+	int level;
+	float roughness;
+} pushConstants;
+layout(set=0, binding=3) uniform sampler textureSampler;
+
+vec4 textureLod(textureCube tex, vec3 p, float lod) {
+	return textureLod(samplerCube(tex, textureSampler), p, lod);
+}
+
+vec2 texSize(textureCube tex, int lod) {
+	return textureSize(samplerCube(tex, textureSampler), lod);
+}
+
+#define PARAM_LEVEL     pushConstants.level
```

Comment:
Since the `#define VULKAN` that necessitated these has been removed, they can be inlined.